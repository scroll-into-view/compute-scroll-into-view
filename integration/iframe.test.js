beforeAll(async () => {
  await page.goto('http://localhost:3000/integration/iframe')
})

describe('scrollable element is "overflow: visible" but hidden by iframe', () => {
  test('should scroll to inside iframe', async () => {
    expect.assertions(3)
    const actual = await page.evaluate(() => {
      const iframe = document.querySelector('iframe')
      const target = iframe.contentDocument.querySelector('.target')
      return window
        .computeScrollIntoView(target, {
          scrollMode: 'always',
        })
        .map(window.mapActions)
    })
    expect(actual).toHaveLength(1)
    expect(actual[0]).toMatchObject({ el: 'html' })
    expect(actual).toMatchSnapshot()
  })
})
