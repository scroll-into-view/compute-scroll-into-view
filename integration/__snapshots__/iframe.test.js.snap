// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrollable element is "overflow: visible" but hidden by iframe should scroll to inside iframe 1`] = `
Array [
  Object {
    "el": "html",
    "left": 0,
    "top": 116,
  },
]
`;
